utils/dependency_cross_matcher.py

```diff
@@ -43,14 +43,31 @@ def main(argv=[], prog_name=''):
     parser = argparse.ArgumentParser(prog='DependencyCrossmatcher',
                                      description='Checks the version mismatches for dependencies '
                                                  'in Cargo based repositories')
+    parser.add_argument("-c", "--compare", action='store_true',
+                        help='Check for mismatches between 2 repositories')
     parser.add_argument('--deps_dir',
                         required=True,
-                        help='Existing directory that contains the Cargo.toml for analyzing'
+                        nargs='+',
+                        help='Existing directories that contain Cargo.toml for analyzing '
                              'dependencies')
     args = parser.parse_args()
 
-    mismatches = run_deps_mismatcher(run_cargo_tree(args.deps_dir))
-    print_deps(mismatches)
```

Comment:
At the beginning of the script in comments can you briefly explain what the script is intended to do and the ways it can be used?

Reply:
This is done in here : https://github.com/parallaxsecond/parsec-tool/pull/116/commits/8defabb54df91a4458b67a2ab0312d9fa500b138#diff-f920291f1970d0b309680ec2e632ad0470e70bf983f097a1b40cc88c066fbb87R43

Should I rephrase it to be clearer or do you want me to duplicate the comment?

This information can also be accessed by calling:
```
python3 utils/dependency_cross_matcher.py --help
usage: DependencyCrossmatcher [-h] [-c] --deps_dir DEPS_DIR [DEPS_DIR ...]

Checks the version mismatches for dependencies in Cargo based repositories

options:
  -h, --help            show this help message and exit
  -c, --compare         Check for mismatches between 2 repositories
  --deps_dir DEPS_DIR [DEPS_DIR ...]
                        Existing directories that contain Cargo.toml for analyzing dependencies
```

Reply:
Yeah, this test block if you can add them as a comment section in the beginning I feel it would be easier for people to understand it rather than going through the code. What do you think?

Reply:
Done. I'm not retriggering the CI Nightly run (that passed) as it's only a comment